server/pkg/integrationapi/item.go

```diff
@@ -56,3 +58,58 @@ func NewItem(i *item.Item, s *schema.Schema, assets *AssetContext) Item {
 		UpdatedAt: lo.ToPtr(i.Timestamp()),
 	}
 }
+
+
+func NewVersionedItemWithPreviousValue(ver item.Versioned ,original item.Versioned, s *schema.Schema, assets *AssetContext) VersionedItemWithPreviousValue {
+	ps := lo.Map(ver.Parents().Values(), func(v version.Version, _ int) types.UUID {
+		return types.UUID(v)
+	})
+	rs := lo.Map(ver.Refs().Values(), func(r version.Ref, _ int) string {
+		return string(r)
+	})
+
+	ii := NewItemWithPreviousValue(ver.Value(), original.Value(), s, assets)
+	return VersionedItemWithPreviousValue{
+		Id:        ii.Id,
+		CreatedAt: ii.CreatedAt,
+		UpdatedAt: ii.UpdatedAt,
+		Fields:    ii.Fields,
+		ModelId:   ii.ModelId,
+		Parents:   &ps,
+		Refs:      &rs,
+		Version:   lo.ToPtr(types.UUID(ver.Version())),
+	}
+}
+
+func NewItemWithPreviousValue(i *item.Item, o *item.Item, s *schema.Schema, assets *AssetContext) ItemWithPreviousValue {
+	fs := lo.FilterMap(i.Fields(), func(f *item.Field, _ int) (FieldWithPreviousValue, bool) {
+		if s == nil {
+			return FieldWithPreviousValue{}, false
+		}
+		sf := s.Field(f.FieldID())
+		if sf == nil {
+			return FieldWithPreviousValue{}, false
+		}
+		ii, _ := lo.Find(o.Fields(), func(field *item.Field) bool {
+			fmt.Println(field.FieldID().Ref());
+			fmt.Println( f.FieldID().Ref());
+			return field.FieldID() == f.FieldID()
+		})
+
+		return FieldWithPreviousValue{
+			Id:    f.FieldID().Ref(),
+			Type:  lo.ToPtr(ToValueType(f.Type())),
+			Value: lo.ToPtr(ToValues(f.Value(), sf.Multiple(), assets)),
+			OldValue: lo.ToPtr(ToValues(ii.Value(), sf.Multiple(), assets)),
+			Key:   util.ToPtrIfNotEmpty(sf.Key().String()),
+		}, true
```

Comment:
this logic always returns the old value even if it's not changed 
for example: if we have two fields like:
```
{
        "type": "text",
        "value": "value 1"
},{
        "type": "text",
        "value": "value 2"
}
```
updating the first field value to `new value1` will have the output
```
{
        "type": "text",
        "value": "value 1",
        "oldValue": "new value1"
},{
        "type": "text",
        "value": "value 2",
        "oldValue": "value2"
}
```